src/Model/Template.php

```diff
@@ -48,9 +45,15 @@ final class Template
     private array $containers = [];
 
     /**
-     * @param string               $name
-     * @param string               $path
-     * @param array<string, mixed> $containers
+     * @param string $name
+     * @param string $path
+     * @param array<string, array{
+     *   name?: string,
+     *   type?: int,
+     *   blocks?: array<string>,
+     *   placement?: Placement,
+     *   shared?: boolean
```

Comment:
```suggestion
     *   shared?: bool
```

Reply:
And this type seems to be used 3 times ; we might declare a type.

Reply:
We can do it but i am not sure, the type is the same as Container but with all properties nullable

Reply:
I see. Best would be to have pseudo-type like `Optional<Container>` which exists in TypeScript

Reply:
Do you know how to do it on phpstan? or thats something that need to be implemented inside phpstan?

Reply:
That doesn't exist in Phpstan atm.